jetty-core/jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RewriteHandler.java

```diff
@@ -123,14 +131,32 @@ public boolean handle(Request request, Response response, Callback callback) thr
             return false;
 
         Rule.Handler input = new Rule.Handler(request);
-        Rule.Handler output = _rules.matchAndApply(input);
+        Rule.Handler result = getRuleContainer().matchAndApply(input);
 
         // No rule matched, call super with the original request.
-        if (output == null)
+        if (result == null)
             return super.handle(request, response, callback);
 
-        // At least one rule matched, call super with the result of the rule applications.
-        output.setHandler(getHandler());
-        return output.handle(output, response, callback);
+        // At least one rule matched, link the last Rule.Handler
+        // to invoke the child Handler of this RewriteHandler.
+        new Output(result, getHandler());
```

Comment:
Also it would be clearer to explicitly link here rather than use side effect within constructor